lib/dep-graph-builders/yarn-lock-v2/build-depgraph-simple.ts

```diff
@@ -76,19 +80,37 @@ const dfsVisit = async (
       node,
     );
 
-    if (!visitedMap.has(childNode.id)) {
-      addPkgNodeToGraph(depGraphBuilder, childNode, {});
-      await dfsVisit(
-        depGraphBuilder,
-        childNode,
-        visitedMap,
-        extractedYarnLockV2Pkgs,
-        strictOutOfSync,
-        includeOptionalDeps,
-        resolutions,
-      );
+    if (localVisited.has(childNode.id)) {
+      if (pruneWithinTopLevel) {
+        const prunedId = `${childNode.id}:pruned`;
+        depGraphBuilder.addPkgNode(
+          { name: childNode.name, version: childNode.version },
+          prunedId,
+          { labels: { scope: node.isDev ? 'dev' : 'prod', pruned: 'true' } },
+        );
```

Comment:
what was wrong with `addPkgNodeToGraph`?

Reply:
I don't think the abstraction is providing value, and wasn't built with nodeId and pkgId being different in mind

Reply:
I can bring it back if need be